src/target/riscv/riscv.c

```diff
@@ -3524,6 +3523,76 @@ COMMAND_HANDLER(riscv_dmi_write)
 	return ERROR_FAIL;
 }
 
+
+COMMAND_HANDLER(riscv_dm_read)
+{
+	if (CMD_ARGC != 1) {
+		LOG_ERROR("Command takes 1 parameter");
+		return ERROR_COMMAND_SYNTAX_ERROR;
+	}
+
+	struct target *target = get_current_target(CMD_CTX);
+	if (!target) {
+		LOG_ERROR("target is NULL!");
+		return ERROR_FAIL;
+	}
+
+	RISCV_INFO(r);
+	if (!r) {
+		LOG_ERROR("riscv_info is NULL!");
+		return ERROR_FAIL;
+	}
+
+	if (r->dm_read) {
+		uint32_t address, value;
+		COMMAND_PARSE_NUMBER(u32, CMD_ARGV[0], address);
+		if (r->dm_read(target, &value, address) != ERROR_OK)
+			return ERROR_FAIL;
+		command_print(CMD, "0x%" PRIx32, value);
+	} else {
+		LOG_ERROR("dm_read is not implemented for this target.");
```

Comment:
I believe you can use LOG_TARGET_ERROR here:

```suggestion
		LOG_TARGET_ERROR(target, "dm_read is not implemented for this target.");
```